plugins/gcp-projects/src/components/ProjectListPage/ProjectListPage.tsx

```diff
@@ -0,0 +1,164 @@
+    <Tooltip title={text}>
+      <span>{text.slice(0, max)}...</span>
+    </Tooltip>
+  );
+};
+
+const labels = (
+  <>
+    <HeaderLabel label="Owner" value="Spotify" />
+    <HeaderLabel label="Lifecycle" value="Production" />
+  </>
+);
+
+const PageContents = () => {
+  const api = useApi(GCPApiRef);
+  const googleApi = useApi(googleAuthApiRef);
+
+  const { loading, error, value } = useAsync(async () => {
+    const token = await googleApi.getAccessToken(
+      'https://www.googleapis.com/auth/cloud-platform.read-only',
+    );
+
+    const projects = api.listProjects({ token });
+    return projects;
+  });
+
+  if (loading) {
+    return <LinearProgress />;
+  }
+
+  if (error) {
+    return (
+      <Typography variant="h2" color="error">
+        {error.message}{' '}
+      </Typography>
+    );
+  }
+
+  return (
+    <Table component={Paper}>
+      <Table aria-label="GCP Projects table">
+        <TableHead>
+          <TableRow>
+            <TableCell>Name</TableCell>
+            <TableCell>Project Number</TableCell>
+            <TableCell>Project ID</TableCell>
+            <TableCell>State</TableCell>
+            <TableCell>Creation Time</TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {value?.map((project: Project) => (
+            <TableRow key={project.projectId}>
+              <TableCell>
+                <Typography>
+                  <LongText text={project.name} max={30} />
+                </Typography>
+              </TableCell>
+              <TableCell>
+                <Typography>
+                  <LongText text={project?.projectNumber || 'Error'} max={30} />
+                </Typography>
+              </TableCell>
+              <TableCell>
+                <Link
+                  to={`project?projectId=${encodeURIComponent(
+                    project.projectId,
+                  )}`}
+                >
+                  <Typography color="primary">
+                    <LongText text={project.projectId} max={60} />
+                  </Typography>
+                </Link>
+              </TableCell>
+              <TableCell>
+                <Typography>
+                  <LongText
+                    text={project?.lifecycleState || 'Error'}
+                    max={30}
+                  />
+                </Typography>
+              </TableCell>
+              <TableCell>
+                <Typography>
+                  <LongText text={project?.createTime || 'Error'} max={30} />
+                </Typography>
+              </TableCell>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+    </Table>
+  );
+};
+
+export const ProjectListPage = () => {
+  return (
+    <Page theme={pageTheme.service}>
+      <Header title="GCP Projects" type="tool">
+        {labels}
+      </Header>
+      <Content>
+        <ContentHeader title="">
+          <Button variant="contained" color="primary" href="/gcp-projects/new">
```

Comment:
i wonder if we should make these like `component={NavLink} to="new"` to not cause page reloads